config/v3_0/types/config_test.go

```diff
@@ -0,0 +1,249 @@
+			},
+			out: errors.ErrPathConflictsSystemd,
+			at:  path.New("json", "storage", "directories", 0, "path"),
+		},
+		// test 3: directory conflicts with systemd unit, error
+		{
+			in: Config{
+				Ignition{},
+				Passwd{},
+				Storage{
+					Directories: []Directory{
+						{
+							Node: Node{Path: "/etc/systemd/system/foo.service"},
+						},
+					},
+				},
+				Systemd{
+					[]Unit{
+						{
+							Name:     "foo.service",
+							Contents: util.StrToPtr("[foo]\nQux=Baz"),
+							Enabled:  util.BoolToPtr(true),
+						},
+					},
+				},
+			},
+			out: errors.ErrPathConflictsSystemd,
+			at:  path.New("json", "storage", "directories", 0, "path"),
+		},
+		// test 4: link conflicts with systemd dropin file, error
+		{
+			in: Config{
+				Ignition{},
+				Passwd{},
+				Storage{
+					Links: []Link{
+						{
+							Node:          Node{Path: "/etc/systemd/system/foo.service.d/bar.conf"},
+							LinkEmbedded1: LinkEmbedded1{Target: "/qux.conf"},
+						},
+					},
+				},
+				Systemd{
+					[]Unit{
+						{
+							Name: "foo.service",
+							Dropins: []Dropin{
+								{
+									Name:     "bar.conf",
+									Contents: util.StrToPtr("[Foo]\nQux=Bar"),
+								},
+							},
+						},
+					},
+				},
+			},
+			out: errors.ErrPathConflictsSystemd,
+			at:  path.New("json", "storage", "links", 0, "path"),
+		},
+		// test 5: non-conflicting scenarios
+		{
+			in: Config{
+				Ignition{},
+				Passwd{},
+				Storage{
+					Files: []File{
+						{
+							Node: Node{Path: "/etc/systemd/system/bar.service.d/baz.conf"},
+						},
+						{
+							Node: Node{Path: "/etc/systemd/system/bar.service"},
+						},
+						{
+							Node: Node{Path: "/etc/systemd/system/foo.service.d/qux.conf"},
+						},
+					},
+					Links: []Link{
+						{
+							Node:          Node{Path: "/etc/systemd/system/qux.service"},
+							LinkEmbedded1: LinkEmbedded1{Target: "/qux.conf"},
+						},
+						{
+							Node:          Node{Path: "/etc/systemd/system/quux.service.d/foo.conf"},
+							LinkEmbedded1: LinkEmbedded1{Target: "/foo.conf"},
+						},
+					},
+					Directories: []Directory{
+						{
+							Node: Node{Path: "/etc/systemd/system/quux.service.d"},
+						},
+					},
+				},
+				Systemd{
+					Units: []Unit{
+						{
+							Name:     "foo.service",
+							Contents: util.StrToPtr("[Foo]\nQux=Baz"),
+							Enabled:  util.BoolToPtr(true),
+						},
+						{
+							Name: "bar.service",
+							Dropins: []Dropin{
+								{
+									Name: "baz.conf",
+								},
+							},
+							Enabled: util.BoolToPtr(true),
+						},
+						{
+							Name: "qux.service",
+							Dropins: []Dropin{
+								{
+									Name:     "bar.conf",
+									Contents: util.StrToPtr("[Foo]\nQux=Baz"),
+								},
+							},
+						},
+						{
+							Name:     "quux.service",
+							Contents: util.StrToPtr("[Foo]\nQux=Baz"),
+							Enabled:  util.BoolToPtr(true),
+						},
+					},
+				},
+			},
+			at: path.New("json", "storage", "files", 0, "path"),
```

Comment:
Copy-paste error?

Reply:
Yup! I forgot to remove it when I changed this to a passing test.